File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.js
const fs = require('fs');
const input = fs.readFileSync(0, 'utf-8').toString().trim().split('\n');
const [N, M] = input[0].split(' ').map(Number);
const board = input.slice(1).map(n => n.split(''));
let min = 64;

for (let n = 0; n <= N - 8; n++) {
    for (let m = 0; m <= M - 8; m++) {
        let count = 0;
        
        for (let dn = n; dn < n + 8; dn++) {
            for (let dm = m; dm < m + 8; dm++) {
                if ((dn + dm) % 2 > 0 && board[dn][dm] === 'W') {
                    count++;
                } else if ((dn + dm) % 2 === 0 && board[dn][dm] === 'B') {
                    count++;
                }
            }
        }
        
        min = Math.min(min, count, 64 - count);
    }
}

console.log(min);

File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.py
import sys

N, M = map(int, input().split())
board = [[1 if x == 'W' else 0 for x in row.rstrip()] for row in sys.stdin.readlines()]
mv = N * M

def validation(r, c):
    count = 0
    
    for i in range(8):
        for j in range(8):
            if (i+j) % 2 != board[i+r][j+c]:
                count += 1
                
    return count

for n in range(N-7):
    for m in range(M-7):
        v = validation(n, m)
        mv = min(v, 64-v, mv)
        
print(mv)